File: classes01/python/thresholding.py
import cv2

# 读入图像
source = cv2.imread("../../data/images/threshold.png",cv2.IMREAD_GRAYSCALE)
cv2.imshow("Original", source)

# 定义阈值和最大值
thresh=0
maxValue=255

# 阈值化处理
ret,source=cv2.threshold(source,thresh,maxValue,cv2.THRESH_BINARY)
cv2.imshow("Threshold",source)

cv2.waitKey()
cv2.destroyAllWindows()

File: classes01/cpp/thresholding.cpp
#include "opencv2/opencv.hpp"
 using namespace std;
 using namespace cv;

 int main(void)
 {
    //  首先读进来一张当通道的灰度图像
    Mat source=imread("../../data/images/threshold.png",IMREAD_GRAYSCALE);

    imshow("Original", source);
    
    int thresh=0;
    int maxValue=255;

    threshold(source,source,thresh,maxValue,THRESH_BINARY);
    // 显示二值化之后的图像
    imshow("Threshold", source);

    waitKey();
    destroyAllWindows();

     return EXIT_SUCCESS;
 }